### PR TITLE
Add fund data as an example

### PR DESCRIPTION
**Pull request by wangershi**
## Description
Add fund data as an example

## Motivation and Context
There are only stock data as the example, we can also use qlib for fund.

## How Has This Been Tested?
A new feature, I test it offline.

## Screenshots of Test Results (if appropriate):
1. Pipeline test:
2. Your own tests:
```
>>> import qlib
>>> from qlib.data import D
>>> qlib.init(provider_uri="~/.qlib/qlib_data/cn_fund_data")
[49788:MainThread](2021-02-28 19:04:43,535) INFO - qlib.Initialization - [config.py:276] - default_conf: client.
[49788:MainThread](2021-02-28 19:04:45,584) WARNING - qlib.Initialization - [config.py:292] - redis connection failed(host=127.0.0.1 port=6379), cache will not be used!
[49788:MainThread](2021-02-28 19:04:49,953) INFO - qlib.Initialization - [__init__.py:46] - qlib successfully initialized based on client settings.
[49788:MainThread](2021-02-28 19:04:49,956) INFO - qlib.Initialization - [__init__.py:47] - data_path=C:\Users\daoz\.qlib\qlib_data\cn_fund_data
>>> df = D.features(D.instruments(market="all"), ["$DWJZ", "$LJJZ"], freq="day")
>>> df
000001     2001-12-18   1.000000   1.000000
           2001-12-21   1.000000   1.000000
           2001-12-28   1.000000   1.000000
           2002-01-04   1.000000   1.000000
           2002-01-11   1.001000   1.001000
...                          ...        ...
000011     2021-02-22  21.035000  27.839001
           2021-02-23  21.070000  27.874001
           2021-02-24  20.670000  27.474001
           2021-02-26  20.219000  27.023001

[16929 rows x 2 columns]
```

## Types of changes

- [ ] Fix bugs
- [x] Add new feature
- [ ] Update documentation

### CODE REVIEW

**Comment by you-n-g** [comment]
@wangershi  Thanks for your PR.
Please  format your code with `python -m black .  -l 120`

**Comment by wangershi** [comment]
> @wangershi Thanks for your PR.
> Please format your code with `python -m black . -l 120`

Done, thanks @you-n-g .

File: scripts/data_collector/utils.py [hunk 1]
```diff
@@ -93,6 +98,78 @@ def _get_calendar(month):
     return calendar
 
 
+def return_date_list(source_dir, date_field_name: str, file_path: Path):
```
Comment (zhupr) [inline scripts/data_collector/utils.py:101]:
```python
def return_date_list(date_field_name: str, file_path: Path):
    date_list = pd.read_csv(file_path, sep=",", index_col=0)[date_field_name].to_list()
    return sorted(map(lambda x: pd.Timestamp(x), date_list))
```

File: scripts/data_collector/utils.py [hunk 2]
```diff
@@ -93,6 +98,78 @@ def _get_calendar(month):
...
+    logger.info(f"get calendar list from {source_dir} by threshold = {threshold}......")
+
+    source_dir = Path(source_dir).expanduser()
+    file_list = list(source_dir.glob("*.csv"))
+
+    _number_all_funds = len(file_list)
+
+    logger.info(f"count how many funds trade in this day......")
+    _dict_count_trade = dict()  # dict{date:count}
+    _fun = partial(return_date_list, source_dir, date_field_name)
```
Comment (zhupr) [inline scripts/data_collector/utils.py:142]: _fun = partial(return_date_list, date_field_name)

File: scripts/data_collector/utils.py [hunk 3]
```diff
@@ -93,6 +98,78 @@ def _get_calendar(month):
...
+    with tqdm(total=_number_all_funds) as p_bar:
+        with ProcessPoolExecutor(max_workers=max_workers) as executor:
```
Comment (zhupr) [inline scripts/data_collector/utils.py:144]: The following code can read the file less：
```python
all_oldest_list = []
with tqdm(total=_number_all_funds) as p_bar:
    with ProcessPoolExecutor(max_workers=max_workers) as executor:
        for date_list in executor.map(_fun, file_list):
            if date_list:
                all_oldest_list.append(date_list[0])
            for date in date_list:
                if date not in _dict_count_trade.keys():
                    _dict_count_trade[date] = 0

                _dict_count_trade[date] += 1

            p_bar.update()

logger.info(f"count how many funds have founded in this day......")
_dict_count_founding = {date: _number_all_funds for date in _dict_count_trade.keys()}  # dict{date:count}
with tqdm(total=_number_all_funds) as p_bar:
    for oldest_date in all_oldest_list:
        for date in _dict_count_founding.keys():
            if date < oldest_date:
                _dict_count_founding[date] -= 1
```

**Comment by wangershi** [comment]
Done, thanks @zhupr .

**Pull request merged**